compiler/native/substitute_test.go

```diff
@@ -94,91 +198,60 @@ func TestNative_SubstituteStages(t *testing.T) {
 					Pull:        "always",
 				},
 			},
+			wantErr: false,
 		},
-	}
-
-	// run test
-	compiler, err := New(c)
-	if err != nil {
-		t.Errorf("Creating compiler returned err: %v", err)
-	}
-
-	got, err := compiler.SubstituteStages(s)
-
-	if err != nil {
-		t.Errorf("SubstituteStages returned err: %v", err)
-	}
-
-	if !reflect.DeepEqual(got, want) {
-		t.Errorf("SubstituteStages is %v, want %v", got, want)
-	}
-}
-
-func TestNative_SubstituteSteps(t *testing.T) {
-	// setup types
-	set := flag.NewFlagSet("test", 0)
-	c := cli.NewContext(nil, set, nil)
-
-	p := yaml.StepSlice{
 		{
-			Commands:    []string{"echo ${FOO}", "echo $${BAR}"},
-			Environment: map[string]string{"FOO": "baz", "BAR": "baz"},
-			Image:       "alpine:latest",
-			Name:        "simple",
-			Pull:        "always",
-		},
-		{
-			Commands:    []string{"echo ${COMPLEX}"},
-			Environment: map[string]string{"COMPLEX": "{\"hello\":\n  \"world\"}"},
-			Image:       "alpine:latest",
-			Name:        "advanced",
-			Pull:        "always",
-		},
-		{
-			Commands:    []string{"echo $NOT_FOUND", "echo ${NOT_FOUND}", "echo $${NOT_FOUND}"},
-			Environment: map[string]string{"FOO": "baz", "BAR": "baz"},
-			Image:       "alpine:latest",
-			Name:        "not_found",
-			Pull:        "always",
-		},
-	}
-
-	want := yaml.StepSlice{
-		{
-			Commands:    []string{"echo baz", "echo ${BAR}"},
-			Environment: map[string]string{"FOO": "baz", "BAR": "baz"},
-			Image:       "alpine:latest",
-			Name:        "simple",
-			Pull:        "always",
-		},
-		{
-			Commands:    []string{"echo \"{\\\"hello\\\":\\n  \\\"world\\\"}\""},
-			Environment: map[string]string{"COMPLEX": "{\"hello\":\n  \"world\"}"},
-			Image:       "alpine:latest",
-			Name:        "advanced",
-			Pull:        "always",
-		},
-		{
-			Commands:    []string{"echo $NOT_FOUND", "echo ${NOT_FOUND}", "echo ${NOT_FOUND}"},
-			Environment: map[string]string{"FOO": "baz", "BAR": "baz"},
-			Image:       "alpine:latest",
-			Name:        "not_found",
-			Pull:        "always",
+			name: "template",
+			args: args{
+				steps: yaml.StepSlice{
+					{
+						Name: "sample",
+						Template: yaml.StepTemplate{
+							Name: "go",
+							Variables: map[string]interface{}{
+								"build_author": "${BUILD_AUTHOR}",
+								"unknown":      "${DEPLOYMENT_PARAMETER_API_IMAGE}",
+							},
+						},
+						Environment: map[string]string{
+							"BUILD_AUTHOR": "testauthor",
+						},
+					},
+				},
+			},
+			want: yaml.StepSlice{
+				{
+					Name: "sample",
+					Template: yaml.StepTemplate{
+						Name: "go",
+						Variables: map[string]interface{}{
+							"build_author": "testauthor",
+							"unknown":      "${DEPLOYMENT_PARAMETER_API_IMAGE}",
+						},
+					},
+					Environment: map[string]string{
+						"BUILD_AUTHOR": "testauthor",
+					},
+				},
+			},
```

Comment:
should we set `wantErr` explicitly like the other test cases?
```
			wantErr: false,
```

Reply:
Resolved.